### DISCUSSION TITLE
Is there any way to change the configuration of port 35672-35682 to the local IP address?

### DISCUSSION THREAD

**Discussion by xyxjxc3** [rabbitmq/rabbitmq-server#11463] [Questions] [answered]
### Is your feature request related to a problem? Please describe.

port 35672-35682 now is an all-zero listening port.,which may cause security risks.

Is there any way to change the configuration of port 35672-35682 to the local IP address?

### Describe the solution you'd like

port 35672-35682 now is an all-zero listening port.,which may cause security risks.

Is there any way to change the configuration of port 35672-35682 to the local IP address?

**Comment by michaelklishin** [comment, marked as answer]
I don't see how a client TCP port can be a "security risk" and how a port can be changed to "a local IP address".

In any case, how client TCP port range can be changed for CLI tools has a [dedicated documentation section](https://www.rabbitmq.com/docs/networking#epmd-inet-dist-port-range).

**Comment by gomoripeti** [comment]
I think the question is related to the fact that the cli tools start a separate erlang node with distribution enabled and they listen for incoming distribution connections (on a port between 35672-35682 by default). And by default they listen on all interfaces (that is what the reporter might refer to as all-zero IP address)

This can be configured by `kernel` `inet_dist_use_interface` (https://www.erlang.org/doc/apps/kernel/kernel_app.html#inet_dist_use_interface)
I think for the CLI tools this can be configured as for example 
`RABBITMQ_CTL_ERL_ARGS="-kernel inet_dist_use_interface '{127,0,0,1}'"`.

And actually the CLI tools don't need to handle any incoming dist connections, its always them which connects to a rabbitmq-server node. So disabling listening is completely fine as well: `RABBITMQ_CTL_ERL_ARGS="-dist_listen false"` (https://www.erlang.org/doc/apps/erts/erl_cmd#dist_listen)

**Reply by lukebakken** [reply]
Thanks, it may make sense to disable listeners by default.
